contracts/LockDealNFT/LockDealNFT.sol

```diff
@@ -58,12 +58,19 @@ contract LockDealNFT is LockDealNFTModifiers, ILockDealNFTEvents {
     function withdraw(
         uint256 poolId
     ) external onlyOwnerOrAdmin(poolId) returns (uint256 withdrawnAmount, bool isFinal) {
-        (withdrawnAmount, isFinal) = IProvider(poolIdToProvider[poolId]).withdraw(poolId);
-        vaultManager.withdrawByVaultId(
-            poolIdToVaultId[poolId],
-            ownerOf(poolId),
-            withdrawnAmount
-        );
+        address provider = poolIdToProvider[poolId];
+        (withdrawnAmount, isFinal) = IProvider(provider).withdraw(poolId);
+        
+        // In case of the bundle pool, skip the withdrawal step
+        (IDealProvierEvents.BasePoolInfo memory poolInfo, ) = IProvider(provider).getData(poolId);
+        if (poolInfo.token != address(0)) {
+            vaultManager.withdrawByVaultId(
+                poolIdToVaultId[poolId],
+                ownerOf(poolId),
+                withdrawnAmount
+            );
+        }
```

Comment:
i think we can make it more general
in case that owner == approved provider
then we do not do the transfers.

that is how we keep the transfer logic here.
in the 25 / 75 example:

the first 2 ids will be owned by the provider = with no transfer
the last one, will do the transfer - the owner is not a provider